src/Kovan-DssSpell.sol

```diff
@@ -44,94 +75,216 @@ contract DssSpellAction is DssAction {
     uint256 constant RAY = 10**27;
     uint256 constant RAD = 10**45;
 
-    address constant MCD_CLIP_YFI_A      = 0x9020C96B06d2ac59e98A0F35f131D491EEcAa2C2; 
-    address constant MCD_CLIP_CALC_YFI_A = 0x54A18C6ceEBDf42D8532EBf5e0a67C430a51b2f6;
-
-    address constant LERP_FAB = 0xa6766Ed3574bAFc6114618E74035C7bb5e9a6aa9;
-
-    function actions() public override {
-        address MCD_VAT = DssExecLib.vat();
-        address MCD_CAT = DssExecLib.cat();
-        address MCD_DOG = DssExecLib.getChangelogAddress("MCD_DOG");
-        address MCD_VOW = DssExecLib.vow();
-        address MCD_END = DssExecLib.end();
-        address MCD_ESM = DssExecLib.getChangelogAddress("MCD_ESM");
-        address CLIPPER_MOM = DssExecLib.getChangelogAddress("CLIPPER_MOM");
-        address ILK_REGISTRY = DssExecLib.getChangelogAddress("ILK_REGISTRY");
-        address PIP_YFI = DssExecLib.getChangelogAddress("PIP_YFI");
-        address MCD_FLIP_YFI_A = DssExecLib.getChangelogAddress("MCD_FLIP_YFI_A");
-
-        // Set CLIP for YFI-A in the DOG
-        DssExecLib.setContract(MCD_DOG, "YFI-A", "clip", MCD_CLIP_YFI_A);
-
-        // Set VOW in the YFI-A CLIP
-        DssExecLib.setContract(MCD_CLIP_YFI_A, "vow", MCD_VOW);
-
-        // Set CALC in the YFI-A CLIP
-        DssExecLib.setContract(MCD_CLIP_YFI_A, "calc", MCD_CLIP_CALC_YFI_A);
-
+    address constant MCD_CLIP_ETH_A       = 0x7dD1Fb6b9aFdBA9F28DB89c81723b8c6B27A2Fbe;
+    address constant MCD_CLIP_CALC_ETH_A  = 0x46bE29C1993d64f0C93e81D69FfAFDF4881806f2;
+    address constant MCD_CLIP_ETH_B       = 0x004676c737FC75A2799dFe745d23F5597620Ad43;
+    address constant MCD_CLIP_CALC_ETH_B  = 0x4672215ADF0556Af60261e97E221c875ce9F0863;
+    address constant MCD_CLIP_ETH_C       = 0x86D5eA244cf6c79227CA73004C963b72431f23ac;
+    address constant MCD_CLIP_CALC_ETH_C  = 0xa8AfB2680cced6de0E1dfe5C35F0FEdFB8E95720;
+    address constant MCD_CLIP_WBTC_A      = 0x5518C2f409Bed4bD5FF3542d9D5002251EEDA892;
+    address constant MCD_CLIP_CALC_WBTC_A = 0x2c39F8C9aE16B84076D7fEA15CE5855925a09DA6;
+
+    function flipperToClipper(Collateral memory col) internal {
+        // Check constructor values of Clipper
+        require(ClipAbstract(col.clipper).vat() == col.vat, "DssSpell/clip-wrong-vat");
```

Comment:
why pass the vat in the struct instead of just calling it directly here? same applies for spotter, dog...

Reply:
Because you will have an external call + SLOAD for each collateral.